_posts/2024-08-06-quarkus-and-leyden.adoc

```diff
@@ -0,0 +1,275 @@
+---
+layout: post
+title: 'Project Leyden'
+date: 2024-08-06
+tags: leyden
+synopsis: 'Project Leyden: exploring its potential for Quarkus users.'
+author: leydenquarkus
+---
+:imagesdir: /assets/images/posts/leyden
+
+You might have heard of https://openjdk.org/projects/leyden/[Project Leyden]: an initiative within the OpenJDK project with very ambitious goals.
+
+When casually namedropping "I'm looking into Project Leyden" with some fellow Java developers these days, I would usually spot some off guard thinking: "Leyden..?", to then offer a hasty save "Ah, right, the project to improve startup times".
+
+Which is not wrong, as indeed startup times improvements are one of its goals, yet we believe the other stated goals of the project offer even greater potential for our favourite platform and its users.
+
+== What is Leyden?
+
+Project Leyden is an initiative from the OpenJDK team. It is an ongoing experiment that is currently being developed by the joint effort of teams from different companies.
```

Comment:
it would be fair to somehow mention its relationship to graalvm.

Reply:
Can you be more precise?

Leyden is not really related to GraalVM in anything other than very general terms.

We do discuss that general relationship below where we talk about GraalVM deciding to do everything AOT vs the JV traditionally doing everything JIT and Leyden adopting a mix-and-match approach.

We also talk about the trade-off Leyden makes vs the trade-ff GraalVM makes near the end and explain why you might pick one over the other.

What more do you think we need to say?

Reply:
I've discussed this with Max yesterday and I think he'll be satisfied with the other changes we're making.